File: swapi-server/src/main/java/de/lkaktus/entities/PlanetDTO.java
package de.lkaktus.entities;

import com.fasterxml.jackson.annotation.JsonProperty;

public record PlanetDTO(
        String climate,
        String created,
        String diameter,
        String edited,
        String gravity,
        String name,
        @JsonProperty("orbital_period") String orbitalPeriod,
        String population,
        @JsonProperty("rotation_period") String rotationPeriod,
        @JsonProperty("surface_water") String surfaceWater,
        String terrain,
        String url,
        String [] films,
        SearchItemDTO [] filmItems,
        String [] residents,
        SearchItemDTO [] residentItems
) {

}

File: swapi-angular/src/app/api/models/PlanetDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { SearchItemDTO } from './SearchItemDTO';

export type PlanetDTO = {
    orbital_period?: string;
    rotation_period?: string;
    surface_water?: string;
    climate?: string;
    created?: string;
    diameter?: string;
    edited?: string;
    gravity?: string;
    name?: string;
    population?: string;
    terrain?: string;
    url?: string;
    films?: Array<string>;
    filmItems?: Array<SearchItemDTO>;
    residents?: Array<string>;
    residentItems?: Array<SearchItemDTO>;
};

